index.src.html

```diff
@@ -830,18 +839,46 @@ <h4 id="algorithm-create" algorithm>Create a `Credential`</h4>
             types in order to support a "sign-up" use case. For the moment, though, we're punting
             on that by restricting the dictionary to a single entry.
 
-    5.  Let |p| be [=a new promise=].
+    6.  Let |p| be [=a new promise=].
 
-    6.  Run the following steps [=in parallel=]:
+    7.  Let |origin| be the [=current settings object=]'s [=environment settings object/origin=].
 
-        1.  Let |r| be the result of executing |interfaces|[0] {{Credential/[[Create]](options)}}
-            internal method on |options|.
+    8.  Run the following substeps [=in parallel=]:
 
-        2.  If |r| is an [=exception=], [=reject=] |p| with |r|.
+        1.  Let |r| be the result of executing |interfaces|[0]'s {{Credential/[[Create]](origin, options)}}
+            internal method given |origin| and |options|.
 
-            Otherwise, [=resolve=] |p| with |r|.
+        2.  If |r| is an [=exception=], [=reject=] |p| with |r|, and terminate these substeps.
 
-    7.  Return |p|.
+        3.  If |r| is a {{Credential}}, [=resolve=] |p| with |r|, and terminate these substeps.
+
+        4.  Otherwise, |r| must be a [=struct=] whose [=items=] are:
+
+            :   <code><dfn>constructResultantCredentialCallback</dfn></code>
+            ::  A Web IDL {{Function}} [=callback function type=] value, whose purpose is to construct,
+                on the main [=event loop=], 
+                an [=interface object=] derived from {{Credential}}, given 
+                <code>[=credentialCreationDataInstance=]</code> and |global|.
+
+                Note: <code>[=credentialCreationDataInstance=]</code>'s functionality is defined on a 
+                    per-[=credential=] type basis by those [=credential=] type specifications needing to
+                    employ it. E.g., see [[WEBAUTHN]].
+
+            :   <code><dfn>credentialCreationDataInstance</dfn></code>
+            ::  An opaque [=struct=] referencing the data necessary to construct the appropriate 
+                type of {{Credential}}-derived object.
+
+        5.  [=Queue a task=] to run <code>|r|.[=constructResultantCredentialCallback=]</code>
+            given <code>|r|.[=credentialCreationDataInstance=]</code> and |global|. 
+            Let |result| be the result.
```

Comment:
Also, the timing here is a bit odd: "Queue a task to X. Y is the result." does some hand-wavey callback magic that we might be able to avoid if we pass the Promise into the callback and let it do the work of resolving/rejecting. That is:

```
5.  Queue a task to run <code>|r|.[=constructResultantCredentialCallback=]</code>
    given <code>|r|.[=credentialCreationDataInstance=]</code>, |global|, and |p|.
```

And then fold 6 and 7 into the callback definition.

Reply:
Ok, great thx, nice polishing.

Reply:
it would be great to get @domenic's feedback on @mikewest's suggestions in [this comment above](https://github.com/w3c/webappsec-credential-management/pull/100/files#r135293896) -- because I am not sure how to declare a data object in an alg in webauthn such that it is still "around" once the callback is invoked, even tho the webauthn alg has terminated itself.

Reply:
I believe we treat variables in spec algorithms as garbage-collected, so you'd just say "let *x* be y" and then use *x* inside the callback steps. As long as nothing changes *x* in parallel with those steps, things are fine.